packages/acceptit-Core.package/ACUserStory.class/class/scenarioAndMethodDict.st

```diff
@@ -7,4 +7,4 @@ scenarioAndMethodDict
 			ifFalse: [nil] "protect from unnormal cirumstances"
 			ifTrue: [cMeth pragmaAt: #scenario])
 				ifNil: ["normal method" md at: cMeth selector put: cMeth; yourself]
-				ifNotNil: [md at: cMeth getSource asString lines first asSymbol put: cMeth; yourself]].
\ No newline at end of file
+				ifNotNil: [md at: (cMeth getSource asString lines first copyReplaceAll: ' ' with: '') asSymbol put: cMeth; yourself]].
```

Comment:
First of all the law of Demeter is hurt badly in line 10. The way I understand line 10 we get the method name from cMeth the entire method? and put it in a new dictionary md. Of what type is cMeth? could we write a getTitleTrimmed? Or we could write class method ACUserstory getTitleFromScenario: CMeth

Reply:
cMeth is a compiledMethod object (duh) so no we can't really add methods to it (or I guess we could but I don't think the teaching team likes star categories for random purposes like this one. CompiledMethod might even have a getter for the selector though which we might be able to use.

Reply:
Further more it took me half an hour to get whats going on. To be honerst some of that is due to my lack of expertise but this code could be more descriptive of its intensions. Why not extract the entire inner part of the block from line 6-10 into a method with descriptive name? like addMethodNameAndContentFrom: cMeth toMethoddictionary: md. And what do you think of renaming md to something sensible, i mean its no markdown file right :-)

Reply:
I just checked what the method actually does (my fix was bandaid I checked what was expected and then what it actually returned and fixed the dictionary to match that exectation). Apparently now I just rebuild the functionality of `methodDict` which is cool because now we can just delete the method :)

Reply:
check the diff now. By removing all that jazz we fix the error aswell and all tests (at least locally still run)